cypress/plugins/index.js

```diff
@@ -12,10 +12,33 @@
 // This function is called when a project is opened or re-opened (e.g. due to
 // the project's config changing)
 
+const secrets = require('../../secrets.applitools.json');
+const getBranchName = require('../../tools/utils/getBranchName');
+
+if (!process.env.APPLITOOLS_API_KEY) {
+  process.env.APPLITOOLS_API_KEY = !!secrets ? secrets.apiKey : '';
+}
+
 /**
  * @type {Cypress.PluginConfig}
  */
 module.exports = (on, config) => {
   // `on` is used to hook into various events Cypress emits
   // `config` is the resolved Cypress config
+
+  if (process.env.GITHUB_BRANCH_NAME != null) {
```

Comment:
How bad would it be if we just rely on the Branch name provided by the GitHub actions context? 

It's not like we have a use case for running locally and if you really need to test it works we can just use a static fall back of `local`

Reply:
Yeah for sure, that makes sense to me. I would love to sync with folks during focus time tomorrow on how we should leverage things like branch name, batch name, etc. 

I'll go ahead and shuffle things around tomorrow to rely only on the name from the GH actions, defaulting to `local` if that is not available. Thanks!